pkg/handler/branch_test.go

```diff
@@ -0,0 +1,30 @@
+package handler
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/gorilla/mux"
+)
+
+func TestGetBranchByTag(t *testing.T) {
+	req, err := http.NewRequest("GET", "/v1/Siddharth-Bhardwaj/dining-concierge/branch/getBranchByTag?tag=1.0.0", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rr := httptest.NewRecorder()
+	router := mux.NewRouter()
+	router.HandleFunc("/v1/{owner}/{repo}/branch/getBranchByTag", NewBranchHandler().GetBranchByTag).Methods("GET")
+
+	router.ServeHTTP(rr, req)
+
+	expectedResponseBody := `{"branches":["master"]}`
+	actualResponseBody := strings.TrimSpace(rr.Body.String())
+
+	if actualResponseBody != expectedResponseBody {
+		t.Errorf("handler returned unexpected body: got %v want %v", actualResponseBody, expectedResponseBody)
+	}
```

Comment:
It would be beneficial to add checks for the status and content type of the response as well

Reply:
good point, i'll add those